### ISSUE TITLE
Do...While , While transformation to For and If

### ISSUE THREAD

**Issue by Konstantin8105** [elliotchance/c2go#191]
In C code we have operators Do..While, While.
Present state of c2go source have implementation of that operators, by in my point of view, we can create better. 
## What I mean?
We can create directly transformation AST tree from Do..While to AST tree of For and If.
## Reason
Typical problem of C code is many obvious and hidden using of comma.
If we save present design of implementation of Do...While, then we have to create a duplicate of code for solving the comma problem for Do..While, While, For, If.
If we change the design to directly transformation of AST trees, then code is free of duplicates.
## Show me
I prepare PR and tested. All is Ok.
## Problem
I can promote PR only after merging PR "COMMA problem for variable".

**Comment by Konstantin8105** [comment]
Fix #201

**Comment by Konstantin8105** [comment]
Fix #202

**Comment by elliotchance** [comment]
Can this issue be closed now that the two PRs are merged in?

**Comment by Konstantin8105** [comment]
Yes, that issue is done.
